circuit/src/op.rs

```diff
@@ -445,7 +461,11 @@ pub trait WitnessHintsFiller<F>: Debug + WitnessFillerClone<F> {
     /// Compute the output given the inputs
     /// # Arguments
     /// * `inputs` - Input witness
-    fn compute_outputs(&self, inputs_val: Vec<F>) -> Result<Vec<F>, CircuitError>;
+    fn compute_outputs(
+        &self,
+        inputs_val: Vec<F>,
```

Comment:
If you add the context you can get rid of `inputs_val`, because they can be derived from the context